File: Pod/july/july16/WeightDifference.java
import java.util.*;

class WeightDifference{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++) arr[i] = sc.nextInt();
        int min = Integer.MAX_VALUE , max = Integer.MIN_VALUE;
        for(int i=0;i<n;i++)
        {
            max = Math.max(max,arr[i]);
            min = Math.min(min,arr[i]);
        }
        System.out.println("Heaviest piece weight: " + max);
        System.out.println("Lightest piece weight: " + min);
        System.out.println("Weight difference: " + (max - min));
    }
}

File: Pod/july/july16/WeightDifference.cpp
#include<iostream>
#include<vector>
#include<climits>

using namespace std;

int main()
{
    int n;
    cin>>n;
    vector<int> arr(n);
    for(int i=0;i<n;i++) cin>>arr[i];
    int min_val = INT_MAX , max_val = INT_MIN;
    for(int x : arr)
    {
        min_val = min(min_val , x);
        max_val = max(max_val , x);
    }
    
    cout<<"Heaviest piece weight: "<< max_val;
    cout<<"\nLightest piece weight: "<< min_val;
    cout<<"\nWeight difference: "<< max_val - min_val;
}